src/estimators/stratified_sketch.py

```diff
@@ -23,53 +23,134 @@
+    that_key_set = set(that.ids().keys())
+    result._ids = {x: 1 for x in result_key_set.union(that_key_set)}
+    return result
+
+  @classmethod
+  def intersection(cls, this, that):
+    """Intersection operation for ExactSet."""
+    if this is None or that is None:
+      return None
+    result = copy.deepcopy(this)
+    result_key_set = set(result.ids().keys())
+    that_key_set = set(that.ids().keys())
+    result._ids = {x: 1 for x in result_key_set.intersection(that_key_set)}
+    return result
+
+  @classmethod
+  def difference(cls, this, that):
+    """Difference operation for ExactSet."""
+    if this is None:
+      return None
+    if that is None:
+      return copy.deepcopy(this)
+    result = copy.deepcopy(this)
+    result_key_set = set(result.ids().keys())
+    that_key_set = set(that.ids().keys())
+    result._ids = {x: 1 for x in result_key_set.difference(that_key_set)}
+    return result
+
+
 class StratifiedSketch(SketchBase):
   """A frequency sketch that contains cardinality sketches per frequency bucket."""
 
   @classmethod
   def get_sketch_factory(cls,
                          max_freq,
                          cardinality_sketch_factory,
-                         underlying_set=None):
+                         noiser_class=None,
+                         epsilon=0,
+                         epsilon_split=0.5,
+                         underlying_set=None,
+                         union=ExactSetOperator.union):
 
     def f(random_seed):
       return cls(
-          random_seed=random_seed,
           max_freq=max_freq,
+          cardinality_sketch_factory=cardinality_sketch_factory,
+          random_seed=random_seed,
           underlying_set=underlying_set,
-          cardinality_sketch_factory=cardinality_sketch_factory)
+          noiser_class=noiser_class,
+          epsilon=epsilon,
+          epsilon_split=epsilon_split,
+          union=union)
 
     return f
 
   def __init__(self,
                max_freq,
                cardinality_sketch_factory,
                random_seed,
-               underlying_set=None):
+               noiser_class=None,
+               epsilon=0,
+               epsilon_split=0.5,
+               underlying_set=None,
+               union=ExactSetOperator.union):
     """Construct a Stratified sketch.
 
     Args:
       max_freq: the maximum targeting frequency level. For example, if it is set
         to 3, then the sketches will include frequency=1, 2, 3+ (frequency >=
         3).
+      cardinality_sketch_factory: A cardinality sketch factory.
       random_seed: This arg exists in order to conform to
         simulator.EstimatorConfig.sketch_factory.
-      cardinality_sketch_factory: A cardinality sketch factory.
+      noiser : A noiser class that is a subclass of base.EstimateNoiserBase.
+      epsilon : Total privacy budget to spend for noising this sketch.
+      epsilon_split : Ratio of privacy budget to spend to noise 1+ sketch.
```

Comment:
Can you add docstring to explain the behavior when epsilon_split=0. also, epsilon_split should be less than 1.